lobsterpy/featurize/batch.py

```diff
@@ -671,19 +672,346 @@ def _get_fingerprints_df(self) -> pd.DataFrame:
+
+        Args:
+            path_to_lobster_calcs: path to root directory consisting of all lobster calc
+            add_additional_data_sg: bool indicating whether to include icoop and icobi data as edge properties
+            which_bonds : selects which kind of bonds are analyzed. "all" is the default
+            start: start energy for bonding antibonding percent integration
+            n_jobs : parallel processes to run
+
+        """
+        self.path_to_lobster_calcs = path_to_lobster_calcs
+        self.add_additional_data_sg = add_additional_data_sg
+        self.which_bonds = which_bonds
+        self.start = start
+        self.n_jobs = n_jobs
+
+    def _get_sg_df(self, path_to_lobster_calc) -> pd.DataFrame:
+        """
+        Generate a structure graph with LOBSTER data bonding analysis data.
+
+        Returns:
+            A  structure graph with LOBSTER data as edge and node properties in structure graph objects
+        """
+        dir_name = Path(path_to_lobster_calc)
+
+        req_files = {
+            "charge_path": "CHARGE.lobster",
+            "cohpcar_path": "COHPCAR.lobster",
+            "icohplist_path": "ICOHPLIST.lobster",
+            "icooplist_path": "ICOOPLIST.lobster",
+            "icobilist_path": "ICOBILIST.lobster",
+            "madelung_path": "MadelungEnergies.lobster",
+            "structure_path": "POSCAR",
+        }
+
+        for file, default_value in req_files.items():
+            file_path = dir_name / default_value
+            req_files[file] = file_path  # type: ignore
+            if not file_path.exists():
+                gz_file_path = Path(zpath(file_path))
+                if gz_file_path.exists():
+                    req_files[file] = gz_file_path  # type: ignore
+
+        charge_path = str(req_files.get("charge_path"))
+        cohpcar_path = str(req_files.get("cohpcar_path"))
+        icohplist_path = str(req_files.get("icohplist_path"))
+        icooplist_path = str(req_files.get("icooplist_path"))
+        icobilist_path = str(req_files.get("icobilist_path"))
+        madelung_path = str(req_files.get("madelung_path"))
+        structure_path = str(req_files.get("structure_path"))
+
+        graph = LobsterGraph(
+            path_to_poscar=structure_path,
+            path_to_charge=charge_path,
+            path_to_cohpcar=cohpcar_path,
+            path_to_icohplist=icohplist_path,
+            add_additional_data_sg=self.add_additional_data_sg,
+            path_to_icooplist=icooplist_path,
+            path_to_icobilist=icobilist_path,
+            path_to_madelung=madelung_path,
+            which_bonds=self.which_bonds,
+            start=self.start,
+        )
+
+        ids = dir_name.name
+
+        df = pd.DataFrame(index=[ids])
+
+        df.loc[ids, "structure_graph"] = graph.sg
+
+        return df
+
+    def get_df(self) -> pd.DataFrame:
+        """
+        Generate a pandas dataframe with structure graph with LOBSTER data.
```

Comment:
Does one need a dataframe for typical applications of the structure graphs?

Reply:
One would not typically need dataframe, I want to keep all the featurizers consistent and having all graphs returned in a single dataframe makes it easier to use them further down the line

Reply:
Okay, good!